### PR TITLE
[ML] Removing exception rethrow in 

### PR DESCRIPTION
**Pull request by jonathan-buttner**
Related to this comment: https://github.com/elastic/elasticsearch/pull/100674#discussion_r1358330688

We don't actually need to call this because we're not doing anything with the response.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/AbstractProcessWorkerExecutorService.java [hunk 1]
```diff
@@ -112,7 +111,6 @@ public void start() {
                     } catch (Exception e) {
                         logger.error(() -> "error handling process [" + processName + "] operation", e);
                     }
-                    EsExecutors.rethrowErrors(ThreadContext.unwrap(runnable));
```
Comment (jonathan-buttner) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/AbstractProcessWorkerExecutorService.java:115]: @davidkyle should I remove references to `preserveContext` too?

Comment (jonathan-buttner) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/AbstractProcessWorkerExecutorService.java:115]: After chatting with Dave R it seems like we should keep the preserved context.

Comment (davidkyle) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/AbstractProcessWorkerExecutorService.java:115]: Yes the code always `unwrap()`. The thread context is the mechanism for preserving headers so we get things like deprecated warnings in the REST API.

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/3rd_party_deployment.yml#L249


However, I think the `EsExecutors.rethrowErrors(` part is redundant and confusing as it only applies to RunnableFutures which are not used here and the InterruptedException is handled in this class. How about adding an assertion that the Runnable is not an instance of RunnableFuture then the `rethrowErrors` can safely be removed

**Comment by jonathan-buttner** [comment]
@elasticmachine merge upstream

**Review by davidkyle — APPROVED** [review]

**Comment by jonathan-buttner** [comment]
Turns out we do get future runnables when `submit` is called (for example here: https://github.com/elastic/elasticsearch/blob/1ca66bde9109bcc716a692be4a3531c0418299e8/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectCommunicator.java#L164) I'm going to revert this change.

**Pull request merged**